test/specs/smtp_spec.rb

```diff
@@ -7,8 +7,8 @@
 BODY     = 'foo'
 
 class Jackal::Mail::Smtp
-  def send_mail(args)
-    $args = args
+  def send_mail(payload, args)
+    payload.set(:smtp_args, args)
```

Comment:
as noted earlier, i was lacking :coffee: and assuming this was actually within the callback. :+1: